### ISSUE TITLE
Constructing many clients error

### ISSUE THREAD

**Issue by arukaen** [kubernetes-sigs/aws-iam-authenticator#308]
Hello all, recently we've enabled aws-iam-authenticator and I'm seeing an error message whenever I run anything that uses `--all-namespaces` (i.e. `kubectl --context cluster get pods --all-namespaces`).

The error message in question is:
```
W0407 12:20:16.744257   32782 exec.go:203] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1001 clients constructed calling "aws-iam-authenticator"
```

kubectl version v1.18.0
aws-iam-auth version: {"Version":"v0.5.0","Commit":"1cfe2a90f68381eacd7b6dcfa2bf689e76eb8b4b"}
OSX: 10.14.6

**Comment by micahhausler** [comment]
This is actually a [warning message](https://github.com/kubernetes/kubernetes/blob/fcc35b046860ab03851b53ff34a10f6ee0cdecf9/staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/https://github.com/kubernetes/kubernetes/blob/fcc35b046860ab03851b53ff34a10f6ee0cdecf9/staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go#L203), and is mainly targeted for resource usage when using client certificate auth. Kubectl/client-go will only invoke `aws-iam-authenticator` once and reuse the token for multiple connections.

**Comment by arukaen** [comment]
Is there a way to turn off this message?

**Comment by micahhausler** [comment]
You might be able to set `--stderrthreshold=ERROR` ([see klog](https://godoc.org/k8s.io/klog)), but if that doesn't work you'd have to file an issue on kubernetes.

**Comment by max-rocket-internet** [comment]
I'm getting about 2000 lines of this error also when running `kubectl` today. There is no `--stderrthreshold` option for `aws-iam-authenticator`. Is there any other fix?

**Comment by max-rocket-internet** [comment]
My errors looked like this:

```
W0902 17:12:19.796919   25579 exec.go:203] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network connections; 1384 clients constructed calling "aws-iam-authenticator"
```

So I just filtered it out with grep, but this is quite hacky:

```
$ kubectl describe pods 2>&1 | grep -v W0902
```

**Comment by sysadmiral** [comment]
This warning appears continuously when running https://github.com/vmware-tanzu/octant

**Comment by dunjoye4real** [comment]
@sysadmiral  I am the same warning using octant. Know of any work around?

**Comment by HeavensRegent** [comment]
/reopen
This is still an issue Nothing has been resolved.

**Comment by philoserf** [comment]
related https://github.com/kubernetes/kubernetes/issues/91913

**Comment by ajcann** [comment]
Also getting this and commented on #91913 with more information.

**Comment by mooreniemi** [comment]
Seeing this (slightly different message so adding for Google) when using `aws` to load data from S3 in `initContainer`:

```
connections; 1377 clients constructed calling "aws"
W1218 16:36:11.240927 1982077 exec.go:271] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network
connections; 1378 clients constructed calling "aws"
W1218 16:36:11.240941 1982077 exec.go:271] constructing many client instances from the same exec auth config can cause performance problems during cert rotation and can exhaust available network
connections; 1379 clients constructed calling "aws"
```

**Comment by irl-segfault** [comment]
I get this when using octant, just FYI.  Wondering if there's a way to force reuse of the same client.

**Comment by TBBle** [comment]
Per https://github.com/kubernetes/kubernetes/issues/91913#issuecomment-840100077 it should be resolved by updating the version of `k8s.io/client-go` being used, which was done for aws-iam-authenticator in #398, released in v0.5.4.

Note that this is a high hit in a search for this error message, but other tools use `k8s.io/client-go` (or just call-out to kubectl) as well, so make sure you're looking at the right project.
